crates/sui-framework/tests/delegation_tests.move

```diff
@@ -193,7 +193,6 @@ module sui::delegation_tests {
             assert!(vector::length(&staked_sui_ids) == 1, 0);
             let staked_sui = test_scenario::take_from_sender_by_id(scenario, *vector::borrow(&staked_sui_ids, 0));
             assert!(staking_pool::staked_sui_amount(&staked_sui) == 50, 0); 
-            assert!(staking_pool::validator_address(&staked_sui) == VALIDATOR_ADDR_2, 0); 
```

Comment:
Nit: can/should this assertion be replaced with a check that the validator address in the `staking_pool_mappings` in the `ValidatorSet` object is equal to `VALIDATOR_ADDR_2`?

More generally, are there any tests that should have the additional check about the correctness of the `staking_pool_mapping`? If not it may be worthwhile writing one to make sure that it stays in-synch.

Reply:
Absolutely. I'm working on testing now.